src/main/java/io/stargate/sgv2/jsonapi/service/embedding/operation/AzureOpenAIEmbeddingClient.java

```diff
@@ -32,10 +28,8 @@
  * href="https://learn.microsoft.com/en-us/azure/ai-services/openai/reference">API reference</a> for
  * details of REST API being called.
  */
-public class AzureOpenAIEmbeddingClient implements EmbeddingProvider {
-  private EmbeddingProviderConfigStore.RequestProperties requestProperties;
-  private String modelName;
-  private int dimension;
+public class AzureOpenAIEmbeddingClient extends EmbeddingProvider {
+  private static final String providerId = ProviderConstants.AZURE_OPENAI;
```

Comment:
Got a suggestion to extract `providerId` and `getErrorMessage()` to abstract class `EmbeddingProvider` in previous PR: https://github.com/stargate/data-api/pull/1159#discussion_r1640393432
But found it was hard since both `providerId` and `getErrorMessage()` would be used in the interface `XXXXEmbeddingProvider` in each class and they should be static. What do you think @tatu-at-datastax ?

Reply:
Typically sub-class would pass `providerId` constant in constructor to super class; and super class would have a field set, using that for its own processing.
I'll have to see what was the thing with `getErrorMessage()`.

It is also fine for abstract base class to define methods for sub-classes to implement, call those methods for its needs.

Reply:
Based on our discussion, this does not quite work like I thought, so we can forget about it for now.